jax/core.py

```diff
@@ -2310,8 +2310,6 @@ def check_map(ctx_factory, prim, in_avals, params):
   if "call_jaxpr" not in params:
     raise JaxprTypeError(f"Map primitive {prim} missing 'call_jaxpr' parameter")
   call_jaxpr = params["call_jaxpr"]
-  if call_jaxpr.effects:
```

Comment:
I don't think we should delete this check entirely. Maybe you want to allow certain effects in maps, but others are still a type error?